gn3/db/rdf/wiki.py

```diff
@@ -180,3 +187,112 @@ def get_comment_history(
     # See note above in the doc-string
     results["data"] = sorted(data, key=lambda d: d["version"], reverse=True)
     return results
+
+
+def get_next_comment_version(
+    comment_id: int, sparql_uri: str, graph: str = "<http://genenetwork.org>"
+) -> int:
+    "Find the next version to add"
+    query = Template(
+        """
+$prefix
+
+SELECT MAX(?version) as ?max_version FROM $graph WHERE {
+    ?comment rdf:type gnc:GNWikiEntry ;
+             dct:identifier "$comment_id"^^xsd:integer ;
+             dct:hasVersion ?version .
+}
+"""
+    ).substitute(prefix=RDF_PREFIXES, graph=graph, comment_id=comment_id)
+    results = sparql_query(
+        query=query, endpoint=sparql_uri, format_type="json")[0]
+    if not results:
+        raise MissingDBDataException
+    return int(results["max_version"]["value"]) + 1
+
+
+def update_wiki_comment(
+    comment_id: int,
+    payload: dict,
+    sparql_conf: dict,
+    graph: str = "<http://genenetwork.org>",
+) -> tuple[str, int]:
+    """Update a wiki comment by inserting a comment with the same
+identifier but an updated version id.  The End form of this query
+looks like:
+
+    INSERT {
+            GRAPH <http://genenetwork.org> {
+            [ rdfs:label '''XXXX'''@en] rdf:type gnc:GNWikiEntry ;
+                    gnt:symbol "XXXX" ;
+                    foaf:mbox <XXXX> ;
+                    gnt:initial "XXXX" ;
+                    gnt:belongsToSpecies ?speciesId ;
+                    gnt:reason "XXXX" ;
+                    foaf:homepage <XXXX> ;
+                    dct:references pmid:XXXX ;
+                    dct:references pmid:XXXX ;
+                    gnt:belongsToCategory "XXXX";
+                    gnt:belongsToCategory "XXXX";
+                    dct:hasVersion "123"^^xsd:integer ;
+                    dct:identifier "1"^^xsd:integer ;
+                    dct:created "2024-09-11 11:00"^^xsd:datetime .
+
+            } USING <http://genenetwork.org> WHERE {
+                    ?speciesId gnt:shortName "{species}" .
+
+            }
+    }
+    """
+    next_version = get_next_comment_version(
+        comment_id, sparql_conf['sparql_uri'], graph)
+    name = f"gn:wiki-{comment_id}-{next_version}"
+    comment_triple = Template("""$name rdf:label '''$comment'''@en ;
+rdf:type gnc:GNWikiEntry ;
+gnt:symbol "$symbol" ;
+dct:identifier "$comment_id"^^xsd:integer ;
+dct:hasVersion "$next_version"^^xsd:integer ;
+dct:created "$created"^^xsd:datetime .
+""").substitute(
+        comment=payload["comment"],
+        name=name, symbol=payload['symbol'],
+        comment_id=comment_id, next_version=next_version,
+        created=datetime.datetime.now(datetime.timezone.utc).strftime('%Y-%m-%d %H:%M:%S'))
```

Comment:
this will be different from the value we store in mysql. Can we make it the same value?